docs/specifications/tests/MethodsV2.md

```diff
@@ -152,42 +174,56 @@ If *Test Type* is "undelegated test", then as specified in method
 
 11. If all servers above are exhausted, then:
-   1. If *Test Type* is "normal test", then return the following from the 
-      method:
-      1. The parent zone as empty (undefined).
-      2. The existence of the *Child Zone* is set to be false.
-      3. The parent name server IP list is set to be empty.
-   2. If *Test Type* is "undelegated test", then return the following from 
-      the method:
-      1. The parent zone as empty (undefined).
-      2. The existence of the *Child Zone* is set to be true.
-      3. The parent name server IP list is set to be empty.
-   3. Exit.
+    1. If *Test Type* is "normal test", then return the following from the
+       method:
+       1. The parent zone as empty (undefined).
+       2. The existence of the *Child Zone* is set to be false.
+       3. The parent name server IP list is set to be empty.
+    2. If *Test Type* is "undelegated test", then return the following from
+       the method:
+       1. The parent zone as empty (undefined).
+       2. The existence of the *Child Zone* is set to be true.
+       3. The parent name server IP list is set to be empty.
+    3. Exit.
+
+### Test Type and existence of parent and child
+
+#### Normal test
+
+Parent zone     | *Child Zone*       | Can normal test be run on *Child Zone*?
+----------------|--------------------|----------------------------------------
+Determined      | Exists             | Yes
+Determined      | Does not exist (1) | No
+Indetermined (2)| Indetermined       | No
+
+1. When parent zone returns an authoritative NXDOMAIN or NODATA on the
+   *Child Zone* name and SOA record.
+2. When server or zone error prevents determination of parent zone.
 
-### Test Type and existence of parent and child 
+#### Undelegated test
 
-Parent zone     | *Child Zone*       | Run normal test? | Run undelegated test?
-----------------|--------------------|------------------|---------------------
-Determined      | Exists             | Yes              | Yes (1)
-Determined      | Does not exist (2) | No               | Yes (3)
-Indetermined (4)| Indetermined       | No               | Yes (3)
+Parent zone     | *Child Zone*       | Can undelegated test be run on *Child Zone*?
+----------------|--------------------|---------------------------------------------
+Determined      | Exists             | Yes (1)
+Determined      | Does not exist (2) | Yes (3)
+Indetermined (4)| Indetermined       | Yes (3)
 
 1. Undelegated tests are run based on the submitted data and not
    the existing zone.
-2. Parent zone returns an authoritative NXDOMAIN or NODATA on the 
+2. When parent zone returns an authoritative NXDOMAIN or NODATA on the
    *Child Zone* name and SOA record.
 3. When *Test Type* is "undelegated test" the *Child Zone* is
    defined to exist even if there is no delegation.
-4. Server or zone error prevents determination of parent zone.
+4. When server or zone error prevents determination of parent zone.
 
-### Outputs
+### Outcome(s)
```

Comment:
> 1. I suggest reverting the change back from "Outcome(s)" to "Outputs". After all this is not a test case. It's a function (in the imperative sense) with inputs and outputs. (Also there are still a couple of headings named "Output" in singular.)

Output or outcome, same, same. It is a simple change so I do it.

>     2. I also suggest to move the section up so it comes just after the "Inputs" section. That way you get the function signature gathered up neatly in one place.

I think it is better to have it after the test procedure when this is not read as code.

>     3. And how about removing the "existance of child zone" output? As you note in the "existence of parent and child" section above, this could be determined by querying the parent zone for the SOA record. Such a check could be done in one of the BASIC test cases and it could fail with a critical message if it gets NXDOMAIN or NODATA. This way the Get-Parent-Zone could be relieved of the Test Type input, and some steps could be removed from the procedure. Or am I missing something?

I changed to true or false to be able to test the zone. 

A new round will be needed to add support for DNAME.

Reply:
>> 2. I also suggest to move the section up so it comes just after the "Inputs" section. That way you get the function signature gathered up neatly in one place.
>
> I think it is better to have it after the test procedure when this is not read as code.

How do you read it when you don't read it as code?

Reply:
In all test case specifications the outcome section (equivalent to the output sections in this specification) is after the test procedures. I see no reason why the order should be different here.